qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -319,159 +313,114 @@ def process_dag(block_dag, wire_map):
-                        mapped_block_dag,
-                        mapped_block_layout,
-                        block_qubit_indices,
-                        (
-                            block_result.result.map,
-                            block_result.result.node_order,
-                            block_result.result.node_block_results,
-                        ),
-                        block_id_to_node,
-                    )
-
-                    # Apply swap epilogue to bring each block to the same
-                    # final layout.
-                    process_swaps(
-                        block_result.swap_epilogue,
-                        mapped_block_dag,
-                        mapped_block_layout,
-                        canonical_register,
-                        False,
-                        bit_to_qreg_idx,
-                        component_map,
-                    )
-
-                    # If the swap epilogue didn't return us to the initial layout,
-                    # there's a bug.
-                    # assert mapped_block_layout.layout_mapping() == current_layout.layout_mapping()
-
-                    mapped_block_dags.append(mapped_block_dag)
-                    idle_qubits.intersection_update(mapped_block_dag.idle_wires())
-
-                mapped_blocks = []
-                for mapped_block_dag in mapped_block_dags:
-                    # Remove wires that are idle in all blocks.
-                    mapped_block_dag.remove_qubits(*idle_qubits)
-                    mapped_blocks.append(dag_to_circuit(mapped_block_dag))
+            empty.add_creg(creg)
+        empty._global_phase = block.global_phase
+        return empty
+
+    def apply_swaps(dest_dag, swaps, layout):
+        for a, b in swaps:
+            # The swaps that come out of Sabre are already in terms of the virtual qubits of the
+            # outermost DAG, since the scope binding occurred as the `SabreDAG` objects were built
+            # up; they're all provided to Sabre routing as full-width already.
+            qubits = (
+                physical_qubits[layout.logical_to_physical(a)],
+                physical_qubits[layout.logical_to_physical(b)],
+            )
+            layout.swap_logical(a, b)
+            dest_dag.apply_operation_back(SwapGate(), qubits, ())
 
-                # Apply the control flow gate to the dag.
-                mapped_node = node.op.replace_blocks(mapped_blocks)
-                mapped_node_qargs = mapped_blocks[0].qubits
-                out_dag.apply_operation_back(mapped_node, mapped_node_qargs, node.cargs)
-                continue
+    def recurse(dest_dag, source_dag, result, root_logical_map, layout):
+        """The main recursive worker.  Mutates ``dest_dag`` and ``layout`` and returns them.
 
-            # If we get here, the node isn't a control-flow gate.
+        ``root_virtual_map`` is a mapping of the (virtual) qubit in ``source_dag`` to the index of
```

Comment:
Shouldn't this be `root_logical_map` to match the argument name?

Reply:
Yes thanks - I use `virtual` to describe this type of qubit, but the Rust side uses `logical` and I ended up mixing the two all over the place.  Fixed in ff33d8d09.